_extensions/diagram/diagram.lua

```diff
@@ -152,17 +152,44 @@ local mermaid = {
     local mime_type = self.mime_type or 'image/svg+xml'
     local file_extension = extension_for_mimetype[mime_type]
     return with_temporary_directory("diagram", function (tmpdir)
-      return with_working_directory(tmpdir, function ()
-        local infile = 'diagram.mmd'
-        local outfile = 'diagram.' .. file_extension
-        write_file(infile, code)
-        pipe(
-          self.execpath or 'mmdc',
-          {"--pdfFit", "--input", infile, "--output", outfile},
-          ''
-        )
-        return read_file(outfile), mime_type
-      end)
+      local infile = tmpdir .. '/diagram.mmd'
```

Comment:
We can use `pandoc.path.join` here to ensure that we are using the correct directory separator. It probably won't matter, but it's good style.